src/labkey/Experiment.ts

```diff
@@ -597,40 +597,64 @@ export function saveRuns(options: SaveRunsOptions): XMLHttpRequest {
     });
 }
 
-export interface GenIdActionsOptions extends RequestCallbackOptions {
+export interface EntitySequenceActionsOptions extends RequestCallbackOptions {
     containerPath?: string;
-    genId?: number;
-    kindName: 'SampleSet' | 'DataClass';
-    rowId: number;
+    seqType: 'genId' | 'rootSampleCount' | 'sampleCount';
+    newValue?: number;
+    kindName?: 'SampleSet' | 'DataClass';
+    rowId?: number;
 }
+
 /**
- * Set the current genId sequence value for the data type (sampleset or dataclass)
+ * Update the current value for the DB Sequences used in sample or data name expression
  */
-export function setGenId(config: GenIdActionsOptions): XMLHttpRequest {
+export function setEntitySequence(config: EntitySequenceActionsOptions): XMLHttpRequest {
     return request({
-        url: buildURL('experiment', 'setGenId.api', config.containerPath),
+        url: buildURL('experiment', 'setEntitySequence.api', config.containerPath),
         method: 'POST',
         success: getCallbackWrapper(getOnSuccess(config), config.scope),
         failure: getCallbackWrapper(getOnFailure(config), config.scope, true),
         jsonData: {
             rowId: config.rowId,
             kindName: config.kindName,
-            genId: config.genId,
+            newValue: config.newValue,
+            seqType: config.seqType,
         },
     });
 }
 
 /**
- * Get the current genId for the data type (sampleset or dataclass)
+ * Set the current genId sequence value for the data type (sampleset or dataclass)
+ */
+export function setGenId(config: EntitySequenceActionsOptions): XMLHttpRequest {
+    return setEntitySequence({
+        ...config,
+        seqType: 'genId',
```

Comment:
The properties here seem a little strange in that this method would seem to require `seqType` even though this method is specifically so you don't have to supply `seqType`, but making `seqType` optional is not accurate since it's needed for `setEntitySequence`. Perhaps have an interface without `seqType` that `EntitySequenceActionsOptions` inherits from?